testutils/testcluster: TestManualReplication failed

testutils/testcluster.TestManualReplication [failed](https://tanzanite.cluster.engflow.com/invocations/default/c687761b-7cee-47eb-a5cf-bcb6a4153377?testReportRun=23&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvdGVzdHV0aWxzL3Rlc3RjbHVzdGVyOnRlc3RjbHVzdGVyX3Rlc3Q=) on master @ [a23be6bce928c3e08074a15815b3c67a657bb40e](https://github.com/cockroachdb/cockroach/commits/a23be6bce928c3e08074a15815b3c67a657bb40e):


```
=== RUN   TestManualReplication
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestManualReplication3124605092
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster_test.go:156: no valid lease
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestManualReplication3124605092
--- FAIL: TestManualReplication (10.72s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=23</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestManualReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45258

testutils/testcluster.TestManualReplication [failed](https://tanzanite.cluster.engflow.com/invocations/default/b67912c1-6ffa-4d88-886c-df99b7ff8c7b?testReportRun=7&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvdGVzdHV0aWxzL3Rlc3RjbHVzdGVyOnRlc3RjbHVzdGVyX3Rlc3Q=) on master @ [49cff91f3501494deaf038671bc643c194a0e3ca](https://github.com/cockroachdb/cockroach/commits/49cff91f3501494deaf038671bc643c194a0e3ca):


```
=== RUN   TestManualReplication
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestManualReplication2606183705
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster_test.go:156: no valid lease
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestManualReplication2606183705
--- FAIL: TestManualReplication (9.70s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestManualReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

testutils/testcluster.TestManualReplication [failed](https://tanzanite.cluster.engflow.com/invocations/default/68ce4107-4848-4b11-b270-660a4fa37421?testReportRun=24&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvdGVzdHV0aWxzL3Rlc3RjbHVzdGVyOnRlc3RjbHVzdGVyX3Rlc3Q=) on master @ [f9df57e2bebd963d10ffd7fa52e4d37cf01b80df](https://github.com/cockroachdb/cockroach/commits/f9df57e2bebd963d10ffd7fa52e4d37cf01b80df):


```
=== RUN   TestManualReplication
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestManualReplication3843382051
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster_test.go:156: no valid lease
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestManualReplication3843382051
--- FAIL: TestManualReplication (10.26s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=24</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestManualReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

testutils/testcluster.TestManualReplication [failed](https://tanzanite.cluster.engflow.com/invocations/default/c64576be-6cad-4dcd-8b43-75193d3a049f?testReportRun=19&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvdGVzdHV0aWxzL3Rlc3RjbHVzdGVyOnRlc3RjbHVzdGVyX3Rlc3Q=) on master @ [bc6d6e05a7c0f9ffd8103740239fdbc83fa78e3f](https://github.com/cockroachdb/cockroach/commits/bc6d6e05a7c0f9ffd8103740239fdbc83fa78e3f):


```
=== RUN   TestManualReplication
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestManualReplication2238223443
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster_test.go:156: no valid lease
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestManualReplication2238223443
--- FAIL: TestManualReplication (9.26s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=19</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestManualReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

testutils/testcluster.TestManualReplication [failed](https://tanzanite.cluster.engflow.com/invocations/default/bb07d113-52bf-4374-bbf7-75d699891b24?testReportRun=22&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvdGVzdHV0aWxzL3Rlc3RjbHVzdGVyOnRlc3RjbHVzdGVyX3Rlc3Q=) on master @ [0b4d620740733ec61cf50ca26d19814299d91f8e](https://github.com/cockroachdb/cockroach/commits/0b4d620740733ec61cf50ca26d19814299d91f8e):


```
=== RUN   TestManualReplication
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestManualReplication737402850
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster_test.go:156: no valid lease
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestManualReplication737402850
--- FAIL: TestManualReplication (10.24s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestManualReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/testutils/testcluster/testcluster_test.go
+++ b/pkg/testutils/testcluster/testcluster_test.go
@@ -131,7 +131,7 @@ func TestManualReplication(t *testing.T) {
 		}
 	}
 
-	// Transfer the lease to node 1.
+	// Transfer the lease to node 2.
 	target := tc.Target(0)
 	leaseHolder, err := tc.FindRangeLeaseHolder(tableRangeDesc, &target)
 	if err != nil {
@@ -150,11 +150,16 @@ func TestManualReplication(t *testing.T) {
 	// Check that the lease holder has changed. We'll use the old lease holder as
 	// the hint, since it's guaranteed that the old lease holder has applied the
 	// new lease.
-	target = tc.Target(0)
-	leaseHolder, err = tc.FindRangeLeaseHolder(tableRangeDesc, &target)
-	if err != nil {
-		t.Fatal(err)
-	}
+	// We wrap this in a SucceedsSoon because N2 might have not applied the lease
+	// yet, it might think that the N1 is the current leaseholder, and the lease
+	// might be INVALID if the time is close to the lease expiration time, or in
+	// the case of leader leases, N2 might not be able to determine the validity
+	// of the lease all together.
+	testutils.SucceedsSoon(t, func() error {
+		leaseHolder, err = tc.FindRangeLeaseHolder(tableRangeDesc, &target)
+		return err
+	})
+
 	if leaseHolder.StoreID != tc.Servers[1].StorageLayer().GetFirstStoreID() {
 		t.Fatalf("expected lease on server idx 1 (node: %d store: %d), but is on node: %+v",
 			tc.Server(1).NodeID(),

